### PR TITLE
Run po4a in pull requests

### PR DESCRIPTION
**Pull request by qwrtln**
Those workflows do three things:
1. Check if `po4a` will change anything, and fail in case if it does.
2. If there is a "po4a check" comment, it will run po4a and report any changes to be committed.
3. If there is a "po4a commit" comment, it will commit po4a changes to the PR's branch. This is limited to members with write access.

### CODE REVIEW

File: .github/workflows/po4a-comment-commit.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      - name: Check if translations require updating
+        run: |
+          git status --porcelain translations
+          if [[ `git status --porcelain translations` ]]; then
+            git status --porcelain translations > files.txt
+          fi
+
+      - name: Add comment
+        run: |
+          if [ -s files.txt ]; then
```
Comment (DarthGandalf) [inline .github/workflows/po4a-comment-commit.yaml:61]: [[?

File: .github/workflows/po4a-comment-commit.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      - name: Run po4a
+        run: |
+          export PO4A_DIR="po4a-${{ env.PO4A_VERSION }}"
+          export PERL5LIB="${PO4A_DIR}/lib:${PERL5LIB}"
+          ./${PO4A_DIR}/po4a -v po4a.cfg
+
+      - name: Check if translations require updating
+        run: |
+          git status --porcelain translations
+          if [[ `git status --porcelain translations` ]]; then
```
Comment (DarthGandalf) [inline .github/workflows/po4a-comment-commit.yaml:55]:
```suggestion
          if git status --porcelain translations; then
```

Or may be the reverse, I don't know what you're trying to do here

File: .github/workflows/po4a-comment-check.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            git status --porcelain translations > files.txt
+          fi
+
+      - name: Add comment
+        run: |
+          if [ -s files.txt ]; then
+            # There are edited files
+            echo '⚠️ `po4a` would edit the following files:' >> body.txt
+            echo '```' >> body.txt
+            cat files.txt >> body.txt
```
Comment (DarthGandalf) [inline .github/workflows/po4a-comment-check.yaml:59]: this will looks simpler:

```
(
  echo foo
  cat bar.txt
) >> body.txt
```

**Review by DarthGandalf — CHANGES_REQUESTED** [review]

File: .github/workflows/po4a-comment-commit.yaml (hunk 2 above)
Comment (qwrtln) [reply .github/workflows/po4a-comment-commit.yaml:55]: I'm checking if the command outputs anything.

File: .github/workflows/po4a-comment-check.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  contents: read
+  issues: read
+  pull-requests: write
+  id-token: 'write'
```
Comment (csagataj2) [inline .github/workflows/po4a-comment-check.yaml:11]:
```suggestion
  id-token: write
```

File: .github/workflows/po4a-comment-commit.yaml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  contents: write
+  issues: read
+  pull-requests: write
+  id-token: 'write'
```
Comment (csagataj2) [inline .github/workflows/po4a-comment-commit.yaml:11]:
```suggestion
  id-token: write
```

File: .github/workflows/po4a-comment-commit.yaml (hunk 2 above)
Comment (csagataj2) [reply .github/workflows/po4a-comment-commit.yaml:55]: I guess the intent was to only create files.txt if `git status --porcelain translations` shows no output

**Review by csagataj2 — CHANGES_REQUESTED** [review]

File: .github/workflows/po4a-comment-commit.yaml (hunk 2 above)
Comment (qwrtln) [reply .github/workflows/po4a-comment-commit.yaml:55]: Ah, sorry, duplicated line.

**Review by csagataj2 — APPROVED** [review]

File: .github/workflows/po4a-comment-commit.yaml (hunk 2 above)
Comment (qwrtln) [reply .github/workflows/po4a-comment-commit.yaml]: Yeah, fixed. That was a leftover from po4a PR check.

**Review by DarthGandalf — APPROVED** [review]

**Pull request merged**